Modules/Relationships/test/Relationships.Domain.Tests/Tests/Aggregates/Relationships/DecomposeRelationshipTests.cs

```diff
@@ -0,0 +1,93 @@
+﻿using Backbone.BuildingBlocks.Domain;
+using Backbone.DevelopmentKit.Identity.ValueObjects;
+using Backbone.Modules.Relationships.Domain.Aggregates.Relationships;
+using Backbone.Modules.Relationships.Domain.Tests.Extensions;
+using Backbone.Tooling;
+using FluentAssertions;
+using Xunit;
+using static Backbone.Modules.Relationships.Domain.Tests.TestHelpers.TestData;
+
+namespace Backbone.Modules.Relationships.Domain.Tests.Tests.Aggregates.Relationships;
+
+public class DecomposeRelationshipTests
+{
+    [Fact]
+    public void Decomposing_relationship_transitions_relationship_to_status_deletion_proposed()
+    {
+        // Arrange
+        var relationship = CreateTerminatedRelationship();
+
+        // Act
+        relationship.Decompose(IDENTITY_2, DEVICE_2);
+
+        // Assert
+        relationship.Status.Should().Be(RelationshipStatus.DeletionProposed);
+    }
+
+    [Fact]
+    public void Decomposing_relationship_creates_an_audit_log_entry()
+    {
+        // Arrange
+        SystemTime.Set("2000-01-01");
+        var relationship = CreateTerminatedRelationship();
+
+        // Act
+        relationship.Decompose(IDENTITY_2, DEVICE_2);
+
+        // Assert
+        relationship.AuditLog.Should().HaveCount(4);
+
+        var auditLogEntry = relationship.AuditLog.Last();
+
+        auditLogEntry.Id.Should().NotBeNull();
+        auditLogEntry.Reason.Should().Be(RelationshipAuditLogEntryReason.Decomposed);
+        auditLogEntry.OldStatus.Should().Be(RelationshipStatus.Terminated);
+        auditLogEntry.NewStatus.Should().Be(RelationshipStatus.DeletionProposed);
+        auditLogEntry.CreatedBy.Should().Be(IDENTITY_2);
+        auditLogEntry.CreatedByDevice.Should().Be(DEVICE_2);
+        auditLogEntry.CreatedAt.Should().Be(DateTime.Parse("2000-01-01"));
+    }
+
+    [Fact]
+    public void Only_the_identity_belonging_to_the_relationship_can_decompose_it()
+    {
+        // Arrange
+        var relationship = CreateTerminatedRelationship();
+        var randomIdentity = IdentityAddress.Create([4, 4, 4], "id4");
```

Comment:
Why don't you use the TestData method to generate a random address?

Reply:
I overlooked the `TestData` method, sorry my bad. Changes made.